src/luarocks/core/cfg.lua

```diff
@@ -735,15 +736,15 @@ function cfg.init(detected, warning)
 
    -- Load system configuration file
    sys_config_file = dir.path(cfg.sysconfdir, config_file_name)
-   local sys_config_ok, err = load_config_file(cfg, platforms, sys_config_file)
+   local sys_config_ok, err = load_config_file(cfg, platforms, sys_config_file, hardcoded)
```

Comment:
Can we do this without changing the `load_config_file` function, but by applying hardcoded values to the result config in case of FORCE_HARDCODED?

Reply:
Changed it